frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx

```diff
@@ -254,26 +259,36 @@ function DashCardInner({
     showClickBehaviorSidebar(dashcard.id);
   }, [dashcard.id, showClickBehaviorSidebar]);
 
-  const changeCardAndRunHandler = useMemo(() => {
-    if (!navigateToNewCardFromDashboard) {
-      return null;
-    }
+  const changeCardAndRunHandler =
+    useCallback<DashCardOnChangeCardAndRunHandler>(
+      ({ nextCard, previousCard, objectId }) => {
+        return navigateToNewCardFromDashboard?.({
+          nextCard,
+          previousCard,
+          dashcard,
+          objectId,
+        });
+      },
+      [dashcard, navigateToNewCardFromDashboard],
+    );
 
-    const handler: DashCardOnChangeCardAndRunHandler = ({
-      nextCard,
-      previousCard,
-      objectId,
-    }) => {
-      navigateToNewCardFromDashboard({
-        nextCard,
-        previousCard,
-        dashcard,
-        objectId,
-      });
-    };
+  const metadata = useSelector(getMetadata);
+  const parameterValues = useSelector(getParameterValues);
 
-    return handler;
-  }, [dashcard, navigateToNewCardFromDashboard]);
+  const href = useMemo(() => {
```

Comment:
I brought it closer to the usage.
It turns out `nextCard` will always be the same as `previousCard` when clicking on title.